### PR TITLE
Routing and admin

### PR DESCRIPTION
**Pull request by danilupion**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -1,20 +1,84 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { HttpClientModule } from '@angular/common/http';
+import { FormsModule } from '@angular/forms';
 import { BrowserModule } from '@angular/platform-browser';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
-import { NgModule } from '@angular/core';
+
+import { MatButtonModule } from '@angular/material/button';
```
Comment (vicmator) [inline src/app/app.module.ts:8]: We could create a new module to export all material module. Here you have a example:

https://github.com/deviatan/horologium/blob/master/frontend/app/modules/material/material.module.ts

File: src/app/components/authDialog/authDialog.component.ts [hunk 2]
```diff
@@ -8,16 +8,13 @@ import { AuthService } from '../../services/api/auth.service';
   styleUrls: ['./authDialog.component.scss'],
 })
 export class AuthDialogComponent {
-  username = '';
-  password = '';
-
   constructor(
     private dialogRef: MatDialogRef<AuthDialogComponent>,
     private apiService: AuthService,
```
Comment (vicmator) [inline src/app/components/authDialog/authDialog.component.ts]: Do you use this service in this component?

**Comment by vicmator** [comment]
Why have you put admin and main-page as independent pages? They both are component too. I would put in components folder as the other components.

File: src/app/components/authDialog/authDialog.component.ts (hunk 2 above)
Comment (danilupion) [reply src/app/components/authDialog/authDialog.component.ts]: Nice catch

File: src/app/app.module.ts (hunk 1 above)
Comment (danilupion) [reply src/app/app.module.ts:8]: As we agreed on the real world ™ maybe in the future

File: src/app/components/header/userMenu/userMenu.component.ts [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-  ) {}
+  @Input()
+  links = [];
+
+  Role = Role;
+
+  constructor(public authService: AuthService) {}
 
   logout() {
     this.authService.logout();
```
Comment (vicmator) [inline src/app/components/header/userMenu/userMenu.component.ts:18]: (IMO) After doing logout, I think you should navigate to home page. For example, if you are in admin page and do logout, you have to exit of admin pages

File: src/app/components/authDialog/authDialog.component.ts [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
 
 @Component({
   selector: 'rm-auth-dialog',
   templateUrl: './authDialog.component.html',
   styleUrls: ['./authDialog.component.scss'],
 })
 export class AuthDialogComponent {
-  username = '';
-  password = '';
+  constructor(private dialogRef: MatDialogRef<AuthDialogComponent>, @Inject(MAT_DIALOG_DATA) public data: any) { }
```
Comment (vicmator) [inline src/app/components/authDialog/authDialog.component.ts:11]: (just a question) Why do you need  `@Inject(MAT_DIALOG_DATA) public data: any`

File: src/app/components/header/userMenu/userMenu.component.html [hunk 5]
```diff
@@ -1,10 +1,16 @@
 <mat-menu #userMenu="matMenu">
+  <ng-template ngFor let-link [ngForOf]="links">
+    <a *ngIf="!link.guard || link.guard()" mat-menu-item [routerLink]="link.url" routerLinkActive="active">
+      <mat-icon *ngIf="link.icon">{{link.icon}}</mat-icon>
+      {{link.text}}
+    </a>
+  </ng-template>
   <button mat-menu-item (click)="logout()">
```
Comment (vicmator) [inline src/app/components/header/userMenu/userMenu.component.html:8]: Why do you use `a` elements for iterated links and `button` for logout button? Could you use the same in both elements?

**Review by vicmator — APPROVED** [review]

File: src/app/components/header/userMenu/userMenu.component.ts (hunk 3 above)
Comment (danilupion) [reply src/app/components/header/userMenu/userMenu.component.ts:18]: Done

File: src/app/components/authDialog/authDialog.component.ts (hunk 4 above)
Comment (danilupion) [reply src/app/components/authDialog/authDialog.component.ts:11]: Technical requirement from material2

File: src/app/components/header/userMenu/userMenu.component.html (hunk 5 above)
Comment (danilupion) [reply src/app/components/header/userMenu/userMenu.component.html:8]: HTML elements have semantics, a is for links, buttons for actions that do not interact with the location

**Pull request merged**